src/pyflex/window_selector.py

```diff
@@ -316,25 +339,28 @@ def select_windows(self):
 
         self.calculate_preliminiaries()
 
+        self.determine_signal_and_noise_indices()
+        if self.config.check_global_data_quality:
+            if not self.check_data_quality():
+                return []
+
```

Comment:
It is actually called in determin_signal_and_noise_indices().

def determine_signal_and_noise_indices(self):
        """
        Calculate the time range of the noise and the signal respectively if
        not yet specified by the user.
        """
        if self.config.noise_end_index is None:
            if not self.ttimes:
                logger.warning("Cannot calculate the end of the noise as "
                               "event and/or station information is not given "
                               "and thus the theoretical arrival times cannot "
                               "be calculated")
            else:
                self.config.noise_end_index = \
                    int(self.ttimes[0]["time"] - self.config.min_period)
        if self.config.signal_start_index is None:
            self.config.signal_start_index = self.config.noise_end_index

        self.config._convert_index(npts=self.observed.stats.npts)